content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -585,6 +585,146 @@ trying to return an object of :ada:`Person` type because :ada:`S'Access`
 would be a dangling reference.
 
 
+.. _Adv_Ada_Self_Reference_Anonymous_Access_Types:
+
+Self-reference
+--------------
+
+Previously, we've seen that we can declare
+:ref:`self-references <Adv_Ada_Self_Reference_Access_Types>` using named access
+types. We can do the same with anonymous access types. Let's revisit the code
+example that implements linked lists:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Self_Reference.Linked_List_Example
+
+    generic
+       type T is private;
+    package Linked_Lists is
+
+       type List is limited private;
+
+       procedure Append_Front
+          (L : in out List;
+           E :        T);
+
+       procedure Append_Rear
+          (L : in out List;
+           E :        T);
+
+       procedure Show (L : List);
+
+    private
+
+       type Component is record
+          Value : T;
+          Next  : access Component;
+          --      ^^^^^^^^^^^^^^^^
+          --       Self-reference
+          --
+          --       (The "Component" type is still
+          --        incomplete at this point.)
```

Comment:
I don't understand the comment (and again at the end of the text). Why is type Component incomplete at that point? I suspect I'm missing something in the Ada rules.